test/unit/controllers/DocumentModalControllerSpec.js

```diff
@@ -34,100 +34,112 @@ describe('DocumentModalController', function () {
+      this.$scope.docID = 'randomdoc'
+
+      this.$controller = _$controller_
+
+      spyOn(this.$rootScope, '$broadcast').and.callThrough()
+      spyOn(this.$scope, '$on').and.callThrough()
+
+      var $scope = this.$scope
+      var $rootScope = this.$rootScope
 
-      $controller = _$controller_
+      this.$controller('DocumentModalController', {
+        $scope: $scope,
+        $rootScope: $rootScope,
+        $modalInstance: $modalInstance,
+        PeersSelectService: PeersSelectService,
+        AppMessagesManager: AppMessagesManager,
+        AppDocsManager: AppDocsManager,
+        AppPeersManager: {},
+        ErrorService: ErrorService
+      })
     })
   })
 
   // define tests
   it('sets the document in the scope', function (done) {
-    createController(false, false)
-
-    expect($docManager.wrapForHistory).toHaveBeenCalledWith($scope.docID)
+    expect(this.AppDocsManager.wrapForHistory).toHaveBeenCalledWith(this.$scope.docID)
     done()
   })
 
   it('forwards a message with a document', function (done) {
-    createController(true, false)
-    $scope.messageID = 'id039'
+    this.$scope.messageID = 'id039'
+    var messageID = this.$scope.messageID
 
-    $scope.forward()
-    expect($input).toEqual({canSend: true})
-    expect($scope.$broadcast).toHaveBeenCalledWith('history_focus', {
+    this.$scope.forward()
+    expect(this.PeersSelectService.$input).toEqual({canSend: true})
+    expect(this.$scope.$broadcast).toHaveBeenCalledWith('history_focus', {
       peerString: 'Peerselected',
       attachment: {
         _: 'fwd_messages',
-        id: [$scope.messageID]
+        id: [messageID]
       }
     })
     done()
   })
 
   it('deletes a message with a document', function (done) {
-    createController(false, false)
-    $scope.messageID = 'id123'
+    this.$scope.messageID = 'id123'
 
-    $scope.delete()
-    expect($input).toEqual({type: 'MESSAGE_DELETE'})
-    expect($messManager.deleteMessages).toHaveBeenCalledWith([$scope.messageID])
+    this.$scope.delete()
+    expect(this.ErrorService.$input).toEqual({type: 'MESSAGE_DELETE'})
+    expect(this.AppMessagesManager.deleteMessages).toHaveBeenCalledWith([this.$scope.messageID])
     done()
   })
 
   it('downloads the document', function (done) {
-    createController(false, false)
-
-    $scope.download()
-    expect($docManager.saveDocFile).toHaveBeenCalledWith($scope.docID)
+    this.$scope.download()
+    expect(this.AppDocsManager.saveDocFile).toHaveBeenCalledWith(this.$scope.docID)
     done()
   })
 
-  it('delete a document linked to a message', function (done) {
-    createController(false, true)
-    $scope.messageID = 'id33'
+  it('can not delete a document not linked to a message', function (done) {
+    this.$scope.messageID = 'id42'
 
-    $rootScope.$broadcast('history_delete')
-    expect($scope.$on).toHaveBeenCalledWith('history_delete', jasmine.any(Function))
-    expect($modalI.dismiss).not.toHaveBeenCalled()
+    var update = {}
```

Comment:
If I recall correctly this is for a $scope.$on with a function: function (e, historyUpdate){....}
I'll change the name